item.inc

```diff
@@ -2099,6 +2114,56 @@ _item_onPlayerUseItem(playerid, Item:id) {
     return CallLocalFunction("OnPlayerUseItem", "dd", playerid, _:id);
 }
 
+_item_pickupSelectItem(playerid, const Item:list[], size = sizeof(list)) {
+    new
+        itemsNearby[MAX_ITEM_NAME * BTN_MAX_INRANGE],
+        itemName[MAX_ITEM_NAME],
+        Item:dummyList[BTN_MAX_INRANGE]; // "Any pointers parameters (references, strings, and arrays) to the caller function are not stored in the closure"
+
+    for (new i; i < size; ++i) {
+        GetItemName(list[i], itemName);
```

Comment:
I can't quite remember but I think this API only gives you the item name, however it would be useful to display the name plus extra information which I think is done via another call.

This is how you render additional details depending on the ItemType - so for weapons, it'll be the ammunition for example.

Reply:
Aren't developers supposed to take care of that themselves? I use `GetItemName` function which runs `OnItemNameRender`. That's where developers are able to append text to the name. See: https://github.com/Southclaws/ScavengeSurvive/blob/master/gamemodes/sss/core/weapon/weapon.pwn#L794-L830
and:
https://github.com/ScavengeSurvive/item/blob/master/item.inc#L1870-L1887

Reply:
Yeah I thought this was a function that did not call `OnItemNameRender` but I was probably thinking of `GetItemTypeName`.